### PR TITLE
[Refresh] Respect cron locks when running crons

### LINKED ISSUE #27: wp cron event run --due-now doesn't respect doing_cron transient
I use `wp cron event run --due-now` as my production cron. Every 5 minutes. But it's a heavy cron.  Sometimes it takes more then 5 minutes. When that happens some events get executed twice. Mostly sending emails.

I would suggest a `--no-doubles` flag or something like that to prevent the cron from doing hooks twice.

When the flag is set, wp-cli should check for the `doing_cron` transient like core does. And set it if it's starting. Like core does.

[Relevant core code](https://github.com/WordPress/WordPress/blob/4.9.6/wp-cron.php#L78)

**Why is it doing things twice**

At the start of a cron all tasks are gathered with [_get_cron_array](https://github.com/WordPress/WordPress/blob/4.9.6/wp-includes/cron.php#L477), events get deleted once the start. And the remaing cron array is still executed. And still in the DB list So after 5 minutes the new cron get also does a [_get_cron_array](https://github.com/WordPress/WordPress/blob/4.9.6/wp-includes/cron.php#L477) and will get all events that the first cron also is still handling (and deleting as it goes).

### PR DESCRIPTION
**Pull request by johnbillion**
This refreshes #38 and excludes the unrelated code quality changes.

Fixes #27.

### CODE REVIEW

**Comment by johnbillion** [comment]
Not sure what's up with the functional tests. Getting many failures with the wp-config.php transformation:

```
001 Scenario: Scheduling and then deleting an event                                                                                                          # features/cron.feature:8
      And I run `wp config set DISABLE_WP_CRON false --raw --type=constant --anchor='// ** MySQL settings - You can get this info from your web host ** //'` # features/cron.feature:5
        $ wp config set DISABLE_WP_CRON false --raw --type=constant --anchor='// ** MySQL settings - You can get this info from your web host ** //'
        
        Error: Could not process the 'wp-config.php' transformation.
        Reason: Unable to locate placement anchor.
        cwd: /tmp/wp-cli-test-run--62f596dd0d4771.02058781/
        run time: 0.21218585968018
        exit status: 1 (RuntimeException)
```

**Comment by wojsmol** [comment]
@johnbillion This is because you are using `--anchor='// ** MySQL settings - You can get this info from your web host ** //` witch don't match `wp-config.php` generated with WP-CLI https://github.com/wp-cli/config-command/blob/f5960976af5b0a6f2f19a1bb9762d84494038759/templates/wp-config.mustache#L22

**Comment by johnbillion** [comment]
Right but that anchor hasn't changed in this PR

**Comment by johnbillion** [comment]
Oh I beg your pardon, yes it has. Hmmmm.

File: features/cron.feature [hunk 1]
```diff
@@ -2,7 +2,8 @@ Feature: Manage WP-Cron events and schedules
 
   Background:
     Given a WP install
-    And I run `wp config set DISABLE_WP_CRON false --raw --type=constant --anchor="if ( ! defined( 'DISABLE_WP_CRON' ) )"`
+    And I run `wp config set DISABLE_WP_CRON false --raw --type=constant --anchor='// ** MySQL settings - You can get this info from your web host ** //'`
```
Comment (wojsmol) [inline features/cron.feature:5]: @johnbillion `--anchor='// ** MySQL settings - You can get this info from your web host ** //'` here is a cause

**Comment by wojsmol** [comment]
Template used to generate `wp-config.php` was updated in https://github.com/wp-cli/config-command/pull/139

File: features/cron.feature (hunk 1 above)
Comment (wojsmol) [reply features/cron.feature:5]: and here is a fix
```suggestion
    And I run `wp config set DISABLE_WP_CRON false --raw --type=constant --anchor='// ** Database settings - You can get this info from your web host ** //'`
```

**Comment by danielbachhuber** [comment]
Proceeding with https://github.com/wp-cli/wp-cli/issues/5594 for this repository. I've captured this PR to https://gist.github.com/danielbachhuber/800870993ce975f7e775219180cc37fa in case this PR is auto-closed or broken in some way.

**Pull request closed without merging**